(Publish 26 June 2024 at 9:00 AM) add browser logs API docs (#17747)
newrelic/docs-website: src/content/docs/browser/new-relic-browser/browser-apis/wraplogger.mdx

tags:
  - Browser
  - Browser monitoring
  - Browser agent and SPA API
metaDescription: Automatically capture data passing through your existing browser logging methods as log events.
redirects:
freshnessValidatedDate: never
---

## Syntax

```js
newrelic.wrapLogger(parent: Object, functionName: string, options?: Object<{ customAttributes?: Object, level?: 'debug|error|info|trace|warn'}>)
```

Automatically captures data passing through existing browser logging methods as log events.

## Requirements

* Browser Pro, or Pro+SPA agent (v1.261.0 or higher)
* If you're using npm to install the browser agent and using a non-standard implementation, you must enable the `logging` feature when instantiating the `BrowserAgent` class. For example, add the following in the`features` array:

  ```js
  import { Logging } from '@newrelic/browser-agent/features/logging'

  const options = {
    info: { ... },
    loader_config: { ... },
    init: { ... },
    features: [
      Logging
    ]
  }
  ```

  For more information, see the [npm browser installation documentation](https://www.npmjs.com/package/@newrelic/browser-agent#new-relic-browser-agent).

## Description

> Upon providing this method with a valid parent container and child function name, the browser agent will record a new `log` event every time the wrapped function is `invoked`, capturing the `first` argument passed to the invoked function as the log's `message`. See the [Logs UI](https://docs.newrelic.com/docs/logs/ui-data/use-logs-ui/) for more information about log events. Optional configurations can be passed along with these captured logs, by use of the `options` argument. Any custom attributes supplied to the API call in the `options` argument (`options.customAttributes`) will be appended as top-level attributes on every log event created by this wrapper. You can control the `logType` of the captured log by supplying a `level` to the `options` argument (`options.level`), which defaults to `info`. **Note** once successfully wrapped, the function's logging detection will not be able to be altered.

Reviewer: @metal-messiah this is another case where it almost reads like it's the browser who's providing the container. Also, could we divide this long sentence into two? Is the proposed revision below accurate?

> Upon providing this method with a valid parent container and child function name, the browser agent will record a new `log` event every time the wrapped function is `invoked`, capturing the `first` argument passed to the invoked function as the log's `message`. See the [Logs UI](https://docs.newrelic.com/docs/logs/ui-data/use-logs-ui/) for more information about log events. Optional configurations can be passed along with these captured logs, by use of the `options` argument. Any custom attributes supplied to the API call in the `options` argument (`options.customAttributes`) will be appended as top-level attributes on every log event created by this wrapper. You can control the `logType` of the captured log by supplying a `level` to the `options` argument (`options.level`), which defaults to `info`. **Note** once successfully wrapped, the function's logging detection will not be able to be altered.

Reviewer: ```suggestion
After you provide this method with a valid parent container and child function name, the browser agent will record a new log event every time the wrapped function is invoked. The first argument is passed to the invoked function as the log's message. See the [Logs UI](https://docs.newrelic.com/docs/logs/ui-data/use-logs-ui/) for more information about log events. 

Optional configurations can be passed along with these captured logs with the `options` argument. Any custom attributes supplied to the API call in the `options` argument (`options.customAttributes`) will be appended as top-level attributes on every log event created by this wrapper. You can control the `logType` of the captured log by supplying a `level` to the `options` argument (`options.level`), which defaults to `info`. Note that once successfully wrapped, the function's logging detection can't be altered.
```
